src/test/kotlin/eli/inevitable/formatters/CellFormatterTest.kt

```diff
@@ -0,0 +1,82 @@
+package eli.inevitable.formatters
+
+import com.thedeanda.lorem.LoremIpsum
+import eli.inevitable.textCell
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import kotlin.test.assertEquals
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CellFormatterTest {
+    private val faker: LoremIpsum = LoremIpsum.getInstance()
+
+    @Test
+    fun `An empty cell is valid`() {
+        val lines = textCell().getLines()
+        assert(lines.isEmpty()) {
+            "Unexpected non-empty cell contents"
+        }
+    }
+
+    @Test
+    fun `A cell with single word is valid`() {
+        val content = faker.getWords(1)
+        val lines = textCell(content).getLines()
+        assertEquals(1, lines.size, """Unexpected number of lines:
+            |   lines=$lines
+            """.trimMargin())
+        assertEquals(content.length, lines.first().length, """Unexpected length of line:
+            |   content=${lines.first()}
+            """.trimMargin())
+    }
+
+    @Test
+    fun `A cell with multiple lines is valid`() {
+        val content = listOf<String>(faker.getWords(1), faker.getWords(1), faker.getWords(1))
+        val formatter = textCell(content)
+        val lines = formatter.getLines()
+        assertEquals(content.size, lines.size, """Unexpected number of lines:
+            |   lines=$lines
+            """.trimMargin())
+        val cellWidth = formatter.getWidth()
+        for(i in 0 until content.size) {
+            assertEquals(cellWidth, lines[i].length, """Unexpected length of content line:
+                |   content=${lines[i]}
+                """.trimMargin())
+        }
+    }
+
+    @Test
+    fun `A cell with vertically filled lines is valid`() {
+        val contentHeight = 5
+        val content = listOf<String>(faker.getWords(1), faker.getWords(1), faker.getWords(1))
+        val formatter = textCell(content) {
+            withTotalHeight(contentHeight)
+        }
+        val lines = formatter.getLines()
+        assertEquals(contentHeight, lines.size, """Unexpected number of lines:
+            |   lines=$lines
+            """.trimMargin())
+        val cellWidth = formatter.getWidth()
+        for(i in 0 until content.size) {
```

Comment:
is there an equivalent spread operator?
```
for (0 .. content.size ) { 

}
```
?

Reply:
Ah, actually yes. But I recently learned there is a better way of doing this too. You can use the repeat function if we don't care about a counting variable. I just forgot to switch all the places over.

Reply:
Oh, I actually am using the counting variable here, so the repeat method wouldn't apply.

BTW, in Kotlin, the double dots specify a range like you are suggesting, but you still have to specify a counting variable. And also the range would explicitly include the last number, and as an index, you frequently want to not include the last item, so that's what the "until" does.